lib/install.js

```diff
@@ -423,6 +424,44 @@ function install (gyp, argv, callback) {
 
 }
 
+function formatHostname (hostname) {
+  // canonicalize the hostname, so that 'oogle.com' won't match 'google.com'
+  return hostname.replace(/^\.*/, '.').toLowerCase()
+}
+
+function parseNoProxyZone (zone) {
+  zone = zone.trim().toLowerCase()
+
+  var zoneParts = zone.split(':', 2)
+  var zoneHost = formatHostname(zoneParts[0])
+  var zonePort = zoneParts[1]
+  var hasPort = zone.indexOf(':') > -1
+
+  return {hostname: zoneHost, port: zonePort, hasPort: hasPort}
+}
+
+function uriInNoProxy (uri, noProxy) {
+  var parsedUri = url.parse(uri);
+  var port = parsedUri.port || (parsedUri.protocol === 'https:' ? '443' : '80')
+  var hostname = formatHostname(parsedUri.hostname)
+  var noProxyList = noProxy.split(',')
+
+  // iterate through the noProxyList until it finds a match.
+  return noProxyList.map(parseNoProxyZone).some(function (noProxyZone) {
+    var isMatchedAt = hostname.indexOf(noProxyZone.hostname)
+    var hostnameMatched = (
+      isMatchedAt > -1 &&
+      (isMatchedAt === hostname.length - noProxyZone.hostname.length)
```

Comment:
Correct me if I'm wrong but isn't this just a circuitous way of saying `hostname === noProxyZone.hostname`?

Reply:
IMO it's a suffix match